src/rt/gtfsrt_update.cc

```diff
@@ -419,11 +427,106 @@ bool update_run(source_idx_t const src,
+
+  auto [r, trip_idx] = gtfsrt_resolve_run(today, tt, &rtt, src, td,
+                                          std::string_view{gtfsrt_trip_id});
+  auto const& rtt_const = rtt;
+  auto const location_seq =
+      r.is_scheduled()
+          ? std::span{tt.route_location_seq_[tt.transport_route_[r.t_.t_idx_]]}
+          : std::span{rtt_const.rt_transport_location_seq_[r.rt_]};
+  auto stops = std::vector<stop::value_type>{};
+
+  for (auto const loc : location_seq) {
+    stops.emplace_back(stop{loc}.value());
+  }
+
+  // add rt_transport if not existent
+  if (!r.is_rt()) {
+    auto const sr = td.schedule_relationship();
+    auto const added_or_replaced =
+        is_added(sr) ||
+        sr == transit_realtime::TripDescriptor_ScheduleRelationship_REPLACEMENT;
+    auto times = added_or_replaced
+                     ? std::vector<delta_t>(location_seq.size() * 2U - 2U, 0)
+                     : std::vector<delta_t>{};
+    r.rt_ = rtt.add_rt_transport(
+        src, tt, r.t_, stops, times, std::string_view{gtfsrt_trip_id},
+        td.route_id(), tt.trip_display_names_[trip_idx].view());
+  }
+
+  // match position to stop
+  auto it = utl::find_if(location_seq, [&](auto const& loc) {
+    auto const stop_loc = stop{loc};
+    auto const stop_position = tt.locations_.get(stop_loc.location_idx());
+    auto const vehicle_position = geo::latlng{vp_lat, vp_lon};
+    return geo::approx_squared_distance(
+               stop_position.pos_, vehicle_position,
+               geo::approx_distance_lng_degrees(vehicle_position)) < 10;
+  });
+  auto stopped_at_idx = std::distance(begin(location_seq), it);
+
+  if (it == end(location_seq)) {
+    log(log_lvl::info, "rt.gtfs.unsupported",
+        R"(Position of Vehicle was not near stop. Skipping Message)", tag,
+        entity.id());
+  }
+
+  // get remaining stops
+  auto const seq_numbers =
+      r.is_scheduled()
+          ? loader::gtfs::
+                stop_seq_number_range{{tt.trip_stop_seq_numbers_[trip_idx]},
+                                      static_cast<stop_idx_t>(
+                                          r.stop_range_.size())}
+          : loader::gtfs::stop_seq_number_range{
+                std::span<stop_idx_t>{},
+                static_cast<stop_idx_t>(location_seq.size())};
+
+  // get delay
+  auto const now =
+      vp.has_timestamp()
+          ? unixtime_t{std::chrono::duration_cast<i32_minutes>(
+                std::chrono::seconds{vp.timestamp()})}
+          : std::chrono::floor<i32_minutes>(std::chrono::system_clock::now());
+  auto const delay =
+      now - tt.event_time(r.t_, stopped_at_idx, event_type::kDep);
+  auto const delay_cast = std::chrono::duration_cast<duration_t>(delay);
+  // update delay for following stops
+  auto seq_it = begin(seq_numbers);
+  for (int i = 0; i <= stopped_at_idx; ++i) {
+    ++seq_it;
+  }
+
+  // update delay for remaining stops
+  for (auto stop_idx = stopped_at_idx + 1; seq_it != seq_numbers.end();
```

Comment:
this doesn't handle transfer_type=4 and block_id transfers correctly?
also: are there any ways this can become a invalid time sequence (has to be monotonously increasing, otherwise the routing won't work properly)
multiple vehicle position messages can be in the GTFS-RT protobuf and this has to be handled gracefully even if it's not a valid/useful update